### PR TITLE
Add remove function

### PR DESCRIPTION
**Pull request by d2lam**
Datastore-base already has remove: https://github.com/screwdriver-cd/screwdriver-datastore-base/blob/master/index.js#L91-L93

We never implemented delete route so didn't need it until now. Adding this so we can use it for plugin-pipelines to delete a pipeline.

### CODE REVIEW

File: lib/imdb.js [hunk 1]
```diff
@@ -20,6 +20,23 @@ function getValue(tableName, key) {
...
 /**
+ * Remove a value from a specific table
+ * @method removeValue
+ * @param  {String} tableName Name of the datastore table
+ * @param  {Number} key       Key of the row to obtain the value from
+ * @return {Object}           The value in the datastore
+ */
+function removeValue(tableName, key) {
+    if (tables[tableName] && tables[tableName][key]) {
+        tables[tableName][key] = null;
```
Comment (nkatzman) [inline lib/imdb.js]: can you called `delete tables[tableName][key]` ? 

Idk what would be better

Comment (d2lam) [inline lib/imdb.js]: Yeah I can. The difference is that the key still exists and its value = null VERSUS the key doesn't exist anymore (if we use delete). Maybe using delete is better.

Comment (nkatzman) [inline lib/imdb.js]: Hm I feel delete is more like an actual remove then

File: lib/imdb.js [hunk 2]
```diff
@@ -20,6 +20,23 @@ function getValue(tableName, key) {
...
+ * @method removeValue
+ * @param  {String} tableName Name of the datastore table
+ * @param  {Number} key       Key of the row to obtain the value from
+ * @return {Object}           The value in the datastore
+ */
+function removeValue(tableName, key) {
+    if (tables[tableName] && tables[tableName][key]) {
+        delete tables[tableName][key];
+
+        return;
```
Comment (FenrirUnbound) [inline lib/imdb.js]: What's the difference between returning `null` and returning `undefined`?

Comment (nkatzman) [inline lib/imdb.js]: You can remove this `return` because you want to return no matter what

File: lib/imdb.js [hunk 3]
```diff
@@ -20,6 +20,23 @@ function getValue(tableName, key) {
 }
 
 /**
+ * Remove a value from a specific table
+ * @method removeValue
+ * @param  {String} tableName Name of the datastore table
+ * @param  {Number} key       Key of the row to obtain the value from
+ * @return {Object}           The value in the datastore
```
Comment (FenrirUnbound) [inline lib/imdb.js]: Update

File: lib/imdb.js [hunk 4]
```diff
@@ -20,6 +20,23 @@ function getValue(tableName, key) {
 }
 
 /**
+ * Remove a value from a specific table
+ * @method removeValue
+ * @param  {String} tableName Name of the datastore table
+ * @param  {Number} key       Key of the row to obtain the value from
```
Comment (FenrirUnbound) [inline lib/imdb.js]: Is it a number?

**Comment by nkatzman** [comment]
👍

**Comment by FenrirUnbound** [comment]
👍 


13opeijzod9hc4

**Comment by petey** [comment]
👍

**Pull request merged**